change-validation/src/main/scala/com/normation/plugins/changevalidation/ChangeRequestJdbcRepository.scala

```diff
@@ -154,8 +154,8 @@ class RoChangeRequestJdbcRepository(
   import doobie.*
 
   // utility method which correctly transform Doobie types towards Box[Vector[ChangeRequest]]
-  private[this] def execQuery(q: Query0[Box[ChangeRequest]]): Box[Vector[ChangeRequest]] = {
-    transactRunBox(xa => {
+  private[this] def execQuery(q: Query0[Box[ChangeRequest]]): IOResult[Vector[ChangeRequest]] = {
+    transactIOResult("conversion from type Doobie to type Box[Vector[ChangeRequest]] has failed")(xa => {
```

Comment:
Since `execQuery` is a method to factorize an operation of `transactIOResult` on a query that may fail, I think the message should always be specified by the caller of `execQuery`, if not the error cause will be too generic and the final error message will end up being confusing. 

So the error message should be as a parameter of the `execQuery` method, and that would also require that every database query/operation have explicit error message, which will improve the consistency and user-friendliness.
For example : 
* `getAll()` should fail with an error message `"Could not select all change request in database"`
* `getByDirective(id, onlyPending)` should fail with an error message `"Could not get change requests for directive ${id}"`

You could define it like so : 
```suggestion
  private[this] def execQuery(errorMsg: String)(q: Query0[Box[ChangeRequest]]): IOResult[Vector[ChangeRequest]] = {
    transactIOResult(errorMsg)(xa => {
```